### ISSUE TITLE
clang-tidy `suspicious-memset-usage` does not warn about `memset(dest, fill_value, 0)`

### ISSUE THREAD

**Issue by cpeterso** [llvm/llvm-project#37446]
|  |  |
| --- | --- |
| Bugzilla Link | [38098](https://llvm.org/bz38098) |
| Version | unspecified |
| OS | All |

## Extended Description 
clang-tidy's `bugprone-suspicious-memset-usage` check warns about `memset(dest, sizeof(int), 0)` but does not warn for `memset(dest, 8, 0)` because the statement does not include `sizeof()`.
```cpp
  #include <string.h>
  void test(int *dest)
  {
    memset(dest, sizeof(int), 0); // WARNING as expected
    memset(dest, 8, 0); // NO WARNING even with hard-coded zero length?!
  }
```
```
clang-tidy --checks="-*,bugprone-suspicious-memset-usage" suspicious-memset-usage.cpp
```
https://clang.llvm.org/extra/clang-tidy/checks/bugprone-suspicious-memset-usage.html

**Comment by LegalizeAdulthood** [comment]
Doesn't seem to be a problem anymore?  I tried reproducing on compiler explorer:

https://godbolt.org/z/fW7WznKfb

**Comment by firewave** [comment]
The compiler (not clang-tidy) warning started with Clang 7.

https://godbolt.org/z/dPG5xvj6q

I wonder if there is just some overlap between `-Wmemset-transposed-args` and `bugprone-suspicious-memset-usage` or if the warning made the clang-tidy check obsolete.

**Comment by vbvictor** [comment]
Yeah, we need to probably deprecate this warning in favour of compiler https://godbolt.org/z/YqrzMbEqo.
